src/server/src/main/java/io/cassandrareaper/management/ICassandraManagementProxy.java

```diff
@@ -195,16 +170,6 @@ AttributeList getAttributes(ObjectName name, String[] attributes)
   // From StreamManagerMBean
   Set<CompositeData> getCurrentStreams();
 
-
-  void addConnectionNotificationListener(NotificationListener listener);
```

Comment:
I understand that some of these methods won't be used/implemented by the Http proxy implementation. The problem is that this interface is used everywhere that the JMX proxy was used. Removing these methods from the interface means that lots of places will now have to reference the Proxy object as a JMX type specifically. I know this PR is likely still a work in progress, but changes to this interface are going to have ripple effects in a lot of the other code that is currently using it.

Reply:
I agree, can't we just make a noop implementation of these methods?
I'd rather keep the interface as is.

Reply:
And those changes are needed, as any method that would use this type of stuff will not work without the JMX connection. Those places that you ran into will need a new implementation.

Reply:
ok, I guess we indeed need separate interfaces for JMX and HTTP specific methods then, right? and the JMX/HTTP implementation classes will also implement each their new interface?

> There's a bit more work to this featureset than I'd say was originally estimated.

Sure

Reply:
Well, I was thinking we would abstract some of those methods, such as MetricsProxy's and make it return those GenericMetrics instead of JMXStat and that sort of stuff.

It'll reduce the surface of the generic interfaces a lot if we stop returning JMX stuff everywhere and instead just make the conversion on the JMX implementation already.